### PR TITLE
chore: prune cruft out of no_op fixture

### PR DESCRIPTION
**Pull request by rb-determined-ai**
The no_op fixture is one of the oldest fixtures we have, and it has built up a lot of cruft.

It is also based on legacy infrastructure that we are trying to remove (SearcherContext and WorkloadSequencer), so we're first removing dead or unnecessary code now, to make the work of rewriting the fixture easier later.

Pointless tests, now removed:
  - test_fail_on_chechpoint_save: added in d0b384ab, used to test the golang workload sequencer, which is long-dead.  This is basically testing the fixture at this point.
  - test_fail_on_preclose_chechpoint_save: added in 2e429f30, also used to test the golang workload sequencer.
  - test_fail_on_first_validation: added in 39726d684ee, also used to test the golang workload sequencer.

Unused configs, now removed:
  - single-in-epochs.yaml
  - single-in-records.yaml
  - adaptive_chaos.yaml

Unused hyperparameters, now removed:
  - validation_secs
  - save_checkpoint_secs
  - load_checkpoint_secs
  - validation_set_size
  - fail_on_first_validation
  - fail_on_chechpoint_save

Weird cruft:
  - the metrics_sigma hparam was exclusively used to crash the trial.  I replaced it with the much clearer "crash_on_startup" hyperparameter.

### CODE REVIEW

File: e2e_tests/tests/cluster/test_exp_continue.py [hunk 1]
```diff
@@ -18,13 +18,13 @@ def test_continue_config_file_cli() -> None:
         sess,
```
Comment (azhou-determined) [inline e2e_tests/tests/cluster/test_exp_continue.py:18]: this test doesn't seem like it adds much more value on top of the one below it. can we just delete it?

Comment (azhou-determined) [reply e2e_tests/tests/cluster/test_exp_continue.py:18]: actually, i just looked at the whole file. most of the tests here seem like overkill... also, we're gonna have to delete/rewrite "continue" tests anyway since post-searcher-context continue will be resume.

that'll be in a separate PR tho so nvm.

File: e2e_tests/tests/experiment/test_core.py [hunk 2]
```diff
@@ -286,30 +286,6 @@ def test_kill_experiment_ignoring_preemption() -> None:
...
-    exp.run_failure_test_with_temp_config(
-        sess,
-        config_obj,
-        conf.fixtures_path("no_op"),
-        error_log,
-    )
-
-
-@pytest.mark.e2e_cpu
-def test_perform_initial_validation() -> None:
```
Comment (azhou-determined) [inline e2e_tests/tests/experiment/test_core.py:304]: why was this deleted? it's def still something we support

File: e2e_tests/tests/fixtures/no_op/model_def.py [hunk 3]
```diff
@@ -52,37 +52,23 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
...
         assert self.num_training_metrics > 0
         self.num_validation_metrics = self.env.hparams.get("num_validation_metrics", 1)
         assert self.num_validation_metrics > 0
-        self.save_secs = self.env.hparams.get("save_checkpoint_seconds", 0)
-        self.load_secs = self.env.hparams.get("load_checkpoint_secs", 0)
         self.metrics_progression = self.env.hparams.get("metrics_progression", "decreasing")
         assert self.metrics_progression in ("increasing", "decreasing", "constant")
         self.metrics_base = self.env.hparams.get("metrics_base", 0.9)
         assert 0 < self.metrics_base < 1
-        self.metrics_sigma = self.env.hparams.get("metrics_sigma", 0.0)
```
Comment (azhou-determined) [inline e2e_tests/tests/fixtures/no_op/model_def.py:80]: lol, what an odd way to get a trial to error out. nice

**Comment by azhou-determined** [comment]
approved, but see perform_initial_validation comment

**Review by azhou-determined — APPROVED** [review]
nice

File: e2e_tests/tests/experiment/test_core.py (hunk 2 above)
Comment (rb-determined-ai) [reply e2e_tests/tests/experiment/test_core.py:304]: it's another test that only tests the workload sequencer, which once upon a time was in the master.  I'll mention this one too in the comment.

The python workload sequencer is about to get pulled out of the noop test, and this should absolutely not be an e2e test.  But since we're actively working to get rid of the wlsq in python and since that file is never touched anyway, I actually don't see the upside of converting this to a python unit test either.

**Comment by rb-determined-ai** [comment]
only failures are known flakes, landing now.

File: e2e_tests/tests/fixtures/no_op/model_def.py [hunk 4]
```diff
@@ -52,37 +52,23 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
...
-        assert 0 <= self.metrics_sigma
         self.write_null = self.env.hparams.get("write_null", False)
 
         self.request_stop = self.env.hparams.get("request_stop", False)
 
+        self.crash_on_startup = self.env.hparams.get("crash_on_startup", False)
```
Comment (jgongd) [inline e2e_tests/tests/fixtures/no_op/model_def.py:71]: I like `crash_on_startup`. `metrics_sigma` was a mystery to me.
